pkg/api/v1/errors.go

```diff
@@ -48,7 +51,12 @@ func ensureValidServerResponse(resp *http.Response) error {
 
 		se.StatusCode = resp.StatusCode
 
-		if err := json.NewDecoder(resp.Body).Decode(&se); err != nil {
+		data, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return err
+		}
+
+		if err := json.Unmarshal(data, &se); err != nil {
```

Comment:
Given this is a change that could easily be regressed, could you add unit tests that ensure that we don't fall back to the previous behavior? I can very easily see that someone is not aware of the Unmarshall/Decode issues and moves back to using a decoder.